Octokit.Tests/Clients/ReleasesClientTests.cs

```diff
@@ -484,6 +486,42 @@ public async Task OverrideDefaultTimeout()
 
                 apiConnection.Received().Post<ReleaseAsset>(Arg.Any<Uri>(), uploadData.RawData, Arg.Any<string>(), uploadData.ContentType, newTimeout);
             }
+
+            [Fact]
+            public async Task CanBeCancelled()
+            {
+                var httpClient = new CancellationTestHttpClient();
+                var connection = new Connection(new ProductHeaderValue("TEST"), httpClient);
+                var apiConnection = new ApiConnection(connection);
+
+                var fixture = new ReleasesClient(apiConnection);
+
+                var release = new Release("https://uploads.github.com/anything");
+                var uploadData = new ReleaseAssetUpload("good", "good/good", Stream.Null, null);
+
+                using (var cts = new CancellationTokenSource())
+                {
+                    var uploadTask = fixture.UploadAsset(release, uploadData, cts.Token);
+
+                    cts.Cancel();
+
+                    await Assert.ThrowsAsync<TaskCanceledException>(() => uploadTask);
+                }
+            }
+
+            private class CancellationTestHttpClient : IHttpClient
+            {
+                public async Task<IResponse> Send(IRequest request, CancellationToken cancellationToken)
+                {
+                    await Task.Delay(TimeSpan.FromSeconds(1), cancellationToken);
+
+                    throw new Exception("HTTP operation was not cancelled");
+                }
+
+                public void Dispose() { }
+
+                public void SetRequestTimeout(TimeSpan timeout) { }
+            }
```

Comment:
Couldn't figure out how to set this up using NSubstitute. Hence, this mock of IHttpClient.

Reply:
I think this is fine 👍🏻

Reply:
fuckd up

- [ ]